README/RCodingStyle.hxx

```diff
@@ -0,0 +1,171 @@
+/// \file RCodingStyle.hxx
+/// \author Axel Naumann <axel@cern.ch>
+/// \date 2018-07-06
+
+/*************************************************************************
+ * Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+// This file demonstrates the coding styles to be used for new ROOT files.
+// See the accompanying source file for context.
+
+// File names correspond to the name of the main class that they contain.
+// Our users will expect that a file called "RCodingStyle.hxx" contains a class
+// named `RCodingStyle`.
+
+// All files start with a Doxygen header, followed by the copyright statement.
+
+// The file must contain no trailing whitespace including no indentation-only lines.
+// Lines breaks should happen between 80 and 120 characters.
+
+// The first non-comment, non-empty line must be the #include guard.
+// Acceptable forms are:
+//   ROOT_RCodingStyle, ROOT_RCodingStyle_hxx, ROOT_RCodingStyle_Hxx,
+//   ROOT_RCODINGSTYLE, ROOT_RCODINGSTYLE_HXX
```

Comment:
What or whom could a difference here confuse? What's the advantage of choosing one? They should be file-local, I sort of like that nobody can rely on them outside this file ;-)

Reply:
We should likely mangle at least the component path in the file. It may become very ugly to debug if the include protector names are duplicate between two unrelated files.

I'd make sense to me to stick with one include protector style (the latter).

Reply:
Choosing only one is good for consistency. Isn't this trying to improve exactly that? I don't really care which one we choose, as long as it's consistent throughout the code. At the very least one choice should be recommended over the others. There's also `#pragma once`, but since it's not in any standard, maybe we should stay away from that.